docs/developer.md

```diff
@@ -31,23 +31,33 @@ guidance on pip installation (e.g., for GPU and TPU support).
 
 ### Building `jaxlib` from source
 
+```{warning}
+While it should typically be possible to compile `jaxlib` from source using
+most modern compilers, the builds are only tested using clang. Pull requests
+are welcomed to improve support for different toolchains, but they are not
```

Comment:
Nit: It's confusing to what "they" refers in this sentence. The pull requests, perhaps?

Just say "other compilers" rather than "they".

Reply:
Done. Much better - thanks!